script/tool/lib/src/gradle_check_command.dart

```diff
@@ -228,24 +239,44 @@ apply plugin: "com.google.cloud.artifactregistry.gradle-plugin"
         r'classpath.*gradle\.plugin\.com\.google\.cloud\.artifactregistry:artifactregistry-gradle-plugin');
     final RegExp artifactRegistryPluginApplyRegex = RegExp(
         r'apply.*plugin.*com\.google\.cloud\.artifactregistry\.gradle-plugin');
+    final RegExp artifactRegistryPluginApplyRegexGP = RegExp(
+        r'id.*com\.google\.cloud\.artifactregistry\.gradle-plugin.*version.*\b\d+\.\d+\.\d+\b');
+    final RegExp artifactRegistryPluginApplyDeclarativeRegex = RegExp(
+        r'\bpluginManagement\b');
+
 
     final bool documentationPresent = gradleLines
         .any((String line) => documentationPresentRegex.hasMatch(line));
     final bool artifactRegistryDefined = gradleLines
         .any((String line) => artifactRegistryDefinitionRegex.hasMatch(line));
     final bool artifactRegistryPluginApplied = gradleLines
         .any((String line) => artifactRegistryPluginApplyRegex.hasMatch(line));
+    final bool declarativeArtifactRegistryApplied = gradleLines
+        .any((String line) => artifactRegistryPluginApplyRegexGP.hasMatch(line));
+    final bool declarativePluginBlockApplied = gradleLines
+        .any((String line) => artifactRegistryPluginApplyDeclarativeRegex.hasMatch(line));
 
-    if (!(documentationPresent &&
-        artifactRegistryDefined &&
-        artifactRegistryPluginApplied)) {
-      printError('Failed Artifact Hub validation. Include the following in '
-          'example root settings.gradle:\n$exampleRootSettingsArtifactHubString');
-    }
+    final bool imperativeArtifactRegistryApplied = artifactRegistryDefined && artifactRegistryPluginApplied;
+
+    final bool validArtifactConfiguration = documentationPresent &&
+        (imperativeArtifactRegistryApplied || declarativeArtifactRegistryApplied);
 
-    return documentationPresent &&
-        artifactRegistryDefined &&
-        artifactRegistryPluginApplied;
+    if (!validArtifactConfiguration) {
+      printError('Failed Artifact Hub validation.');
+      if (!documentationPresent) {
+        printError('The link to the Artifact Hub documentation is missing. Include the following in '
+            'example root settings.gradle:\n// See $artifactHubDocumentationString for more info.');
+      }
+      if (artifactRegistryDefined || artifactRegistryPluginApplied || !declarativePluginBlockApplied) {
+        printError('Include the following in '
+            'example root settings.gradle:\n$exampleRootSettingsArtifactHubString');
+      }
+      else if (!declarativeArtifactRegistryApplied){
```

Comment:
I think this file needs `dart format` run on it. (CI currently doesn't check formatting for script/tool, which is a bug.)

Reply:
done